src/hpc/atomic.h

```diff
@@ -13,6 +13,7 @@
 #else
 
 #ifndef WARD_ENABLED
+#define AO_AO_TS_T 0    // HACK
```

Comment:
Ah, indeed, there should be a comment here -- I forgot about this, my bad. I'll update the PR, but to add it here quickly: This was meant to silence `-Wundef` warnings in libatomic_ops. But this is actually problematic, because it will cause issues on some non-x86 architectures like ARM.

I could instead patch our copy of libatomic. We'd then get the warnings again once we add support for using stock versions of libatomic_ops... but I guess that's acceptable (and we can revisit the issue then)...

I guess I could also try something like:
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wundef"
#include <atomic_ops.h>
#pragma GCC diagnostic pop
```
And something similar for clang.. Hum